app/src/main/java/com/mybus/activity/DisplayFavoritesActivity.java

```diff
@@ -20,4 +103,90 @@ public int getToolbarId() {
+                        oldFavorite = mFavorites.get(mFavoritePositionToEdit);
+                        favoriteNameAlertDialog.setPreviousName(oldFavorite.getName());
+                        break;
+                }
+                favoriteNameAlertDialog.show(getFragmentManager(), "Favorite Name");
+        }
+    }
+
+    @Override
+    public void onFavoriteItemEdit(int position) {
+        this.mFavoritePositionToEdit = position;
+        startSearchActivityForFavorite(EDIT_SEARCH_RESULT_ID, mFavorites.get(position).getAddress());
+    }
+
+    @Override
+    public void onFavoriteItemDelete(final int position) {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(this.getResources().getString(R.string.favorite_confirm_delete))
+                .setCancelable(false)
+                .setPositiveButton(this.getResources().getString(R.string.yes), new DialogInterface.OnClickListener() {
+                    public void onClick(final DialogInterface dialog, final int id) {
+                        Long favid = mFavorites.get(position).getId();
+                        //check if the favorite was successfully removed form the database
+                        if (FavoriteLocationDao.getInstance(DisplayFavoritesActivity.this).remove(favid)) {
+                            //remove it locally
+                            mFavorites.remove(position);
+                            //if there are no favorites, show the no favorites message
+                            if(mFavorites.isEmpty()){
+                                mNoFavoritesTextView.setVisibility(View.VISIBLE);
+                            }
+                        }
+                        mFavoriteViewAdapter.notifyItemRemoved(position);
+                    }
+                })
+                .setNegativeButton(this.getResources().getString(R.string.no), new DialogInterface.OnClickListener() {
+                    public void onClick(final DialogInterface dialog, final int id) {
+                        dialog.cancel();
+                    }
+                });
+        final AlertDialog alert = builder.create();
+        alert.show();
+    }
+
+    @Override
+    public void onAddNewFavorite(String favoriteName) {
+        FavoriteLocation newFavorite = new FavoriteLocation();
+        newFavorite.setName(favoriteName);
+        newFavorite.setAddress(newAddress);
+        newFavorite.setLatitude(newLocation.latitude);
+        newFavorite.setLongitude(newLocation.longitude);
+        if (FavoriteLocationDao.getInstance(this).saveOrUpdate(newFavorite)) {
+            mFavorites.add(newFavorite);
+            mFavoriteViewAdapter.notifyDataSetChanged();
+            //hide the no favorites message
+            mNoFavoritesTextView.setVisibility(View.GONE);
+        }
+    }
+
+    @Override
+    public void onChangeFavoriteName(String newName) {
```

Comment:
This method seems to be missing the part of saving the Favorite Location into the Database. With this addition add and change callbacks will look the same, we can refactor this into only one listener.